src/Sandbox.MqttHive/Hive.cs

```diff
@@ -0,0 +1,158 @@
+﻿using HiveMQtt.Client;
+using HiveMQtt.Client.Events;
+using HiveMQtt.Client.Options;
+using HiveMQtt.MQTT5.Types;
+using System.Collections.Concurrent;
+using System.ComponentModel.Design.Serialization;
+using System.Diagnostics;
+
+namespace Sandbox.MqttHive;
+public class Hive
+{
+    public ConcurrentDictionary<string, Bee> Bees { get; } = [];
+    public bool IsConnected => _client.IsConnected();
+    public TimeSpan MaxIdleTime { get; set; } = TimeSpan.FromSeconds(5);
+
+    public Hive(HiveMQClientOptions options)
+    {
+        _client = new HiveMQClient(options);
+    }
+
+    public async Task Initialise()
+    {
+        _client.OnMessageReceived += OnMessageReceived;
+
+        var result = await _client.ConnectAsync();
+
+        if (!IsConnected)
+        {
+            throw new Exception($"Failed to connect to broker, reason: {result.ReasonString}");
+        }
+
+        var token = new CancellationToken();
+        _ = Task.Run(async () => await PingBees(token));
+    }
+
+    public async Task GetBees()
+    {
+        if (!IsConnected)
+        {
+            throw new Exception("Hive is not connected to broker!");
+        }
+
+        var isListing = _client.Subscriptions.Any(subscription => subscription.TopicFilter.Topic == Topics.HiveRoot);
+        if (!isListing)
+        {
+            await _client.SubscribeAsync(Topics.HiveRoot, QualityOfService.AtMostOnceDelivery);
+        }
+
+        // Assuming all bee's are disconnected until they reply
```

Comment:
```suggestion
        // Assuming all bees are disconnected until they reply
```
`bee's` = the thing belongs to 1 bee
`bees` = more than 1 bee

Reply:
yeah my brain just does bee's==bees when in typing mode. same with your and you're. I do know the difference 😆

Reply:
That's why you should proof read your own code.